File: gpt-3.5-ft-large/stage2/acc_memcpy_device.c
#include "acc_testsuite.h"
#ifndef T1
//T1: acc memcpy device,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    real_t *a = (real_t *)malloc(n * sizeof(real_t));
    real_t *b = (real_t *)malloc(n * sizeof(real_t));
    real_t *c = (real_t *)malloc(n * sizeof(real_t));
    real_t *d_a = (real_t *)malloc(n * sizeof(real_t));
    real_t *d_b = (real_t *)malloc(n * sizeof(real_t));
    real_t *d_c = (real_t *)malloc(n * sizeof(real_t));
    int data_err;

    for (int x = 0; x < n; ++x){
        a[x] = rand() / (real_t)(RAND_MAX / 10);
        b[x] = rand() / (real_t)(RAND_MAX / 10);
        c[x] = 0.0;
    }

    #pragma acc enter data create(c[0:n])

    for (int x = 0; x < n; ++x){
        d_c[x] = c[x];
    }
    acc_deviceptr_t dev_ptr = acc_copyin(c, n * sizeof(real_t));
    d_c = dev_ptr;
    acc_detach(c);

    #pragma acc data copyin(a[0:n], b[0:n])
    {
        #pragma acc parallel
        {
            #pragma acc loop
            for (int x = 0; x < n; ++x){
                d_c[x] += a[x] + b[x];
            }
        }
        #pragma acc exit data copyout(c[0:n])
    }

    #pragma acc wait

    data_err = 0;
    for (int x = 0; x < n; ++x){
        if (fabs(c[x] - (a[x] + b[x]))) {
            data_err += 1;
        }
    }
    #pragma acc exit data delete(d_a[0:n], d_b[0:n], c[0:n])
    
    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/stage2/acc_memcpy_device.f90
#ifndef T1
!T1:acc memcpy device,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        !Initializing Data for testing
        INTEGER, DIMENSION(10) :: host_data
        INTEGER, DIMENSION(10) :: dev_data
        INTEGER, DIMENSION(10) :: new_data
        INTEGER :: x

        SEEDDIM(1) = 1
	SEEDDIM(2) = 1
#ifdef SEED
        SEEDDIM(1) = SEED
#endif
        CALL RANDOM_SEED(PUT=SEEDDIM)
        CALL RANDOM_NUMBER(host_data)
        
        !$acc data copyin(host_data(1:10)) copy(dev_data(1:10))
          !$acc parallel
            !$acc loop
            DO x = 1, 10
              dev_data(x) = host_data(x)
            END DO
          !$acc end parallel
        !$acc end data

        CALL acc_memcpy_devicr(new_data, dev_data, 10 * sizeof(INTEGER))

        DO x = 1, 10
          IF (host_data(x) .ne. new_data(x)) THEN
            errors = errors + 1
          END IF
        END DO


        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM